### ISSUE TITLE
Ensure templates consistent with migration

### ISSUE THREAD

**Issue by mihlema** [Spine-project/SpineOpt.jl#464]
During our last developer call, we discussed that we should find a way to ensure that templates are kept up-to-date. 
Possibly, if a migration incorporates changes in a certain parameter definition or class name, we could 
- run the new migration script on each template file before merging (this would imply, that migration scripts would need to be able to accept `.json` as inputs as well)
- enforce a unit test that fails if the template does not correspond to the latest migration version
Let me know what you think.

**Comment by jkiviluo** [comment]
Would be very good development - both of those.

**Comment by DillonJ** [comment]
This is a good idea - especially the second bullet, a test to check that the template corresponds to latest version. I think we should include this in milestone 1.0

**Comment by clizbe** [comment]
@tarskul Is this maybe relevant to our updates in documentation, tutorials, etc? I think you mentioned templates as well.

**Comment by tarskul** [comment]
@clizbe Yeah,. Though, so far we only consider the unit tests for the tutorials (which will also be with JSON files).
